bindings/rust/src/bindings/mod.rs

```diff
@@ -211,6 +273,61 @@ impl Blob {
     pub fn from_hex(hex_str: &str) -> Result<Self, Error> {
         Self::from_bytes(&hex_to_bytes(hex_str)?)
     }
+
+    pub fn samples_and_proofs(
+        &self,
+        kzg_settings: &KZGSettings,
+    ) -> Result<Vec<(Sample, KZGProof)>, Error> {
+        let samples_len = kzg_settings.max_width as usize;
+        let proofs_len = kzg_settings.sample_count as usize;
+
+        let mut samples = Vec::with_capacity(samples_len);
+        let mut proofs = Vec::with_capacity(proofs_len);
+
+        unsafe {
+            let res = get_samples_and_proofs(
+                samples.as_mut_ptr(),
+                proofs.as_mut_ptr(),
+                self,
+                kzg_settings,
+            );
+            if res != C_KZG_RET::C_KZG_OK {
+                return Err(Error::CError(res));
+            }
+            samples.set_len(samples_len);
+            proofs.set_len(proofs_len);
+        }
+
+        let samples = samples
+            .chunks_exact(kzg_settings.sample_size as usize)
+            .map(|felts| Sample::from_field_elements(felts, kzg_settings).unwrap());
+        let ret = samples.zip(proofs).collect();
+        Ok(ret)
+    }
+}
+
+pub fn two_dim_samples<const ROWS: usize, const COLS: usize>(
```

Comment:
I think we should rename this to `get_2d_samples`.

Though, it's actually `get_2d_samples_and_proofs` in the latest version of the `das` branch.